Update installation documentation to align with latest changes (#233)
SimpleDataLabsInc/prophecy-docs: docs/architecture/deployment/installation-guide.md

## Installation Instructions

As mentioned above, Prophecy is installed via Helm chart called the Prophecy-installer. The following are the commonly used values which can be configured for this helm chart. The helm chart values can be broadly classified into 5 different sections

1. **Athena:** Configurations which are used to configure the management plane.

> 1. **Global:** Configurations which are used to configure all the common values between different components like Athena, Postgres and Platform components.

Reviewer: Numbers in the list are all 1s.

1. **Postgres:** Is the section used to configure various postgres/database level configurations used by Prophecy.
1. **Platform:** Is the section used to configure the various platform components like - elastic-search, promethus, grafana, loki etc.
1. **Version:** The Prophecy version to be deployed

:::warning

Parameters marked with an asterisk(\*) are mandatory.

:::

| Parameter                                  | Type                     | Description                                                                                                                                                                                                                      | Default value                       |
| ------------------------------------------ | ------------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------- |

> | **ATHENA**                                 |

Reviewer 2: Good to not to keep esp. default password in public doc

| `athena.config.athenaEnvs`                 | map[string]string        | Provide the values you wish to override in athena ENV variables. Passed in the format of `KEY: "value"`. Value is type sensitive. This field is only respected when fromFile is set to `false`.                                  | `./configs/athena-env-cluster.yaml` |
| `athena.config.envs`                       | map[string]string        | Provide the values you wish to override in Prophecy application ENV variables. Passed in the format of `KEY: "value"`. Value is type sensitive. This field is only respected when fromFile is set to `false`.                    | `./configs/env-cluster.yaml`        |
| `athena.config.prophecyCluster`            | prophecyCluster (custom) | Provide the values you wish to override in Prophecy cluster spec. This field is only respected when fromFile is set to `false`.                                                                                                  | `./configs/prophecy-cluster.yaml`   |

> | `athena.controlcenter`                     | -                        |                                                                                                                                                                                                                                  |                                     |

Reviewer 2: Lets not publish control center ,may be we can tell if there is req

> | `athena.controlcenter.disabled`            | bool                     | Set to `true` if cluster is unable to connect to controlcenter.                                                                                                                                                                  | `false`                             |

Reviewer: Description: Instead of unable to connect to controlcenter, you can mention "if you don't wish to connect to controlcenter for sharing monitoring statistics with Prophecy."

Author: ```suggestion
| `athena.controlcenter.disabled`            | bool                     | if connectivity to controlcenter is not required                                                                                                                                                                  | `false`                             |
```

| `athena.resources`                         | -                        |                                                                                                                                                                                                                                  |                                     |
| `athena.resources.requests`                | -                        |                                                                                                                                                                                                                                  |                                     |
| `athena.resources.requests.cpu`            | string                   | Guaranteed value of CPU required to deploy Athena                                                                                                                                                                                | `200m`                              |

> | `athena.resources.requests.memory`         | string                   | Guaranteed value of CPUs required to deploy Athena                                                                                                                                                                               | `512Mi`                             |

Reviewer: Description: Guaranteed value of memory

| `global.ingressController.type`            | string                   | Type of ingress controller being used. Currently we support `nginx` and `istio`                                                                                                                                                  | `nginx`                             |
| `global.ingressController.class`           | string                   | Name of the controller class which is to be used for ingress resources                                                                                                                                                           | `prophecy-nginx`                    |
| `global.istio`                             | -                        |                                                                                                                                                                                                                                  |                                     |

> | `global.istio.enabled`                     | bool                     | Set this to true to allows usage of istio as an ingress controller/gateway. We don't support sidecars yet. Istio isn't installed or managed by Prophecy.                                                                         | `false`                             |

Reviewer: Description: Set this to true to allow usage of

| `global.istio.gateway.create`              | bool                     | Set this to true, when you wish to create the Istio gateway                                                                                                                                                                      | `false`                             |
| `global.prophecy`                          | -                        |                                                                                                                                                                                                                                  |                                     |
| `global.prophecy.domainValue`              | string                   | domain value used for cookies. Used to configure multiple URLs.                                                                                                                                                                  | `""`                                |

> | `global.prophecy.envSecret`                | -                        | Is configred when Prophecy credentials are fetched as a kubernetes secret                                                                                                                                                        |                                     |

Reviewer: Description: Is configured

| `global.prophecy.envSecret.data`           | map[string]string        | Provide the values you wish to override in Prophecy secret ENV variables. Passed in the format of `METADATA_DB_PASSWORD: dummyPassword123`.                                                                                      |                                     |
| `global.prophecy.envSecret.name`           | string                   | Name of the Kubernetes secret to be used.                                                                                                                                                                                        | `env-secrets`                       |
| `global.prophecy.imagePullSecret`          | string                   | Name of the Kubernetes secret which contains the image pull credentials to the container registry.                                                                                                                               | `""`                                |

> | `global.prophecy.rootUrl`\*                | string                   | Root URL host used to identify the DNS traffic. You may provider a list of comma separated urls.                                                                                                                                 | `""`                                |

Reviewer: Description: Root URL where the Prophecy deployment will be hosted. You may provider a list of comma separated urls if you wish to work with multiple urls. Note that the first url in the provided list will be considered as the primary url.

| `global.prophecy.wildcardCert`             | -                        |                                                                                                                                                                                                                                  |                                     |

> | `global.prophecy.wildcardCert.name`        | string                   | List of comma separated secret cert names.                                                                                                                                                                                       | `""`                                |

Reviewer: Description: List of comma separated secret cert names. The number of secrets passed should either be one or equal to the number of rootUrls provided which should be generated for each of these URLs in respective order.

Author: ```suggestion
| `global.prophecy.wildcardCert.name`        | string                   | The number of secrets passed should either be one or equal to the number of rootUrls provided which should be generated for each of these URLs in respective order.                                                                                                                                                                            | `""`                                |
```

| `global.repository`                        | string                   | Container registry prefix to be used.                                                                                                                                                                                            | `gcr.io/prophecy-share`             |
| `global.tls`                               | -                        |                                                                                                                                                                                                                                  |                                     |
| `global.tls.enabled`                       | bool                     | Specifies if TLS is to be enabled                                                                                                                                                                                                | `true`                              |

> | `global.tls.certOnLB`                      | bool                     | Specifies wheather TLS termination is to be done at the loadbalancer                                                                                                                                                             | `false`                             |

Reviewer: *Specifies whether

| **POSTGRES**                               |

> | `postgres.isExternalPostgres`              | bool                     | Setting this to true, will allow Prophecy applications to use a externally managed Postgres instance and will not use a Prophecy managed instance.                                                                               | `false`                             |

Reviewer: *will allow Prophecy services to use an externally managed

Author: ```suggestion
| `postgres.isExternalPostgres`              | bool                     | Setting this to true, will allow Prophecy services to use a externally managed Postgres instance and will not use a Prophecy managed instance.                                                                               | `false`                             |
```

| `postgres.resources`                       | -                        |                                                                                                                                                                                                                                  |                                     |
| `postgres.resources.requests`              | -                        |                                                                                                                                                                                                                                  |                                     |
| `postgres.resources.requests.cpu`          | string                   | Guaranteed value of CPU required to deploy Postgres                                                                                                                                                                              | `1`                                 |

> | `postgres.resources.requests.memory`       | string                   | Guaranteed value of CPUs required to deploy Postgres                                                                                                                                                                             | `2000Mi`                            |

Reviewer: Guaranteed value of memory

Author: ```suggestion
| `postgres.resources.requests.memory`       | string                   | Guaranteed value of memory required to deploy Postgres                                                                                                                                                                             | `2000Mi`                            |
```

| **PLATFORM**                               |
| `platform.enabled`                         | bool                     | Specifies if one/more of the platform components are to be enabled                                                                                                                                                               | `true`                              |
| `platform.elasticsearch.enabled`           | bool                     | Specifies if elasticsearch is to be enabled and deployed for supporting Prophecy search feature                                                                                                                                  | `true`                              |
| `platform.ingressNginx.enabled`            | bool                     | Specifies if nginx controller is to be deployed for ingress routing                                                                                                                                                              | `true`                              |

> | `platform.loki.enabled`                    | bool                     | Specifies if loki is to be enabled for log collection                                                                                                                                                                            | `false`                             |

Reviewer 2: Loki/promtail/prometheus options we can remove , wil make it by default we dont install

> | `platform.elasticsearch.enabled`           | bool                     | Specifies if elasticsearch is to be enabled                                                                                                                                                                                      | `true`                              |

Reviewer 2: looks its repeated in line 125
